# LinkR - plugin Prescription

https://ohdsi.github.io/TheBookOfOhdsi/StandardizedVocabularies.html

In [1]:
# Librairies:

import pandas as pd
import seaborn as sns
import matplotlib as plt
import plotly.io as pio
import plotly.express as px
import plotly.graph_objs as go
from plotly.subplots import make_subplots


# Drug Exposure

In [2]:
drug_exposure = pd.read_csv('Data/drug_exposure.csv')
print(drug_exposure.shape) #(18229, 23)
'''
['drug_exposure_id', 'person_id', 'drug_concept_id',
       'drug_exposure_start_date', 'drug_exposure_start_datetime',
       'drug_exposure_end_date', 'drug_exposure_end_datetime',
       'verbatim_end_date', 'drug_type_concept_id', 'stop_reason', 'refills',
       'quantity', 'days_supply', 'sig', 'route_concept_id', 'lot_number',
       'provider_id', 'visit_occurrence_id', 'visit_detail_id',
       'drug_source_value', 'drug_source_concept_id', 'route_source_value',
       'dose_unit_source_value']
'''
drug_exposure.person_id.value_counts()
'''
person_id
 7155255168997124770    1182
 4783904755296699562     791
 8090044958540695372     716
-3908355835367628651     687
 7131048714591189903     644
                        ... 
-2575767131279873665      37
-4183220989401122518      34
 1740609625029317924      26
 3912882389848878631      24
-8769042030325953499      18
'''
drug_exposure.info()
'''
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 18229 entries, 0 to 18228
Data columns (total 23 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   drug_exposure_id              18229 non-null  int64  
 1   person_id                     18229 non-null  int64  
 2   drug_concept_id               18229 non-null  int64  
 3   drug_exposure_start_date      18229 non-null  object 
 4   drug_exposure_start_datetime  18229 non-null  object 
 5   drug_exposure_end_date        18229 non-null  object 
 6   drug_exposure_end_datetime    18229 non-null  object 
 7   verbatim_end_date             0 non-null      float64
 8   drug_type_concept_id          18229 non-null  int64  
 9   stop_reason                   0 non-null      float64
 10  refills                       0 non-null      float64
 11  quantity                      18220 non-null  float64
 12  days_supply                   0 non-null      float64
 13  sig                           0 non-null      float64
 14  route_concept_id              18229 non-null  int64  
 15  lot_number                    0 non-null      float64
 16  provider_id                   0 non-null      float64
 17  visit_occurrence_id           18229 non-null  int64  
 18  visit_detail_id               0 non-null      float64
 19  drug_source_value             18229 non-null  object 
 20  drug_source_concept_id        18229 non-null  int64  
 21  route_source_value            18223 non-null  object 
 22  dose_unit_source_value        18220 non-null  object 
 '''

drug_exposure.head()

liste_non_null = ['drug_exposure_id', 'person_id', 'drug_concept_id',  'drug_exposure_start_date', 'drug_exposure_start_datetime',
       'drug_exposure_end_date', 'drug_exposure_end_datetime', 'drug_type_concept_id',   'quantity',
       'route_concept_id', 'visit_occurrence_id', 'drug_source_value', 'drug_source_concept_id', 'route_source_value', 'dose_unit_source_value']


(18229, 23)
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 18229 entries, 0 to 18228
Data columns (total 23 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   drug_exposure_id              18229 non-null  int64  
 1   person_id                     18229 non-null  int64  
 2   drug_concept_id               18229 non-null  int64  
 3   drug_exposure_start_date      18229 non-null  object 
 4   drug_exposure_start_datetime  18229 non-null  object 
 5   drug_exposure_end_date        18229 non-null  object 
 6   drug_exposure_end_datetime    18229 non-null  object 
 7   verbatim_end_date             0 non-null      float64
 8   drug_type_concept_id          18229 non-null  int64  
 9   stop_reason                   0 non-null      float64
 10  refills                       0 non-null      float64
 11  quantity                      18220 non-null  float64
 12  days_supply                   0 non-null      fl

In [57]:
drug_exposure.dose_unit_source_value.unique()
'''
#array(['VIAL', 'mL', 'TUBE', 'AERO', 'TAB', 'UDCUP', 'SYR', 'BAG', 'PKT',
       'SUPP', 'CAP', 'g', 'KIT', 'INH', 'PTCH', 'BTL', 'HALF TAB', 'DEV',
       'mg', 'AMP', 'NEB', 'DRP', nan, 'SPRY', 'dose', 'DBTL', 'LOZ',
       'JAR', 'SYRP', 'UNIT', 'STCK', 'PEN', 'CADD', 'Appl', 'WAF', 'CAN',
       'TROC', 'CART'], dtype=object)
'''

array(['VIAL', 'mL', 'TUBE', 'AERO', 'TAB', 'UDCUP', 'SYR', 'BAG', 'PKT',
       'SUPP', 'CAP', 'g', 'KIT', 'INH', 'PTCH', 'BTL', 'HALF TAB', 'DEV',
       'mg', 'AMP', 'NEB', 'DRP', nan, 'SPRY', 'dose', 'DBTL', 'LOZ',
       'JAR', 'SYRP', 'UNIT', 'STCK', 'PEN', 'CADD', 'Appl', 'WAF', 'CAN',
       'TROC', 'CART'], dtype=object)

# CONCEPT:

Match medicine (concept_name, concept_id) with the drug_concept_id from drug exposure

In [ ]:
con = pd.read_csv(f'/Data/CONCEPT.csv', sep='\t')
#print(con.shape) #(516708, 10)

'''
con[con.concept_id == 40232756]
#      concept_id	concept_name	                        domain_id	vocabulary_id	concept_class_id	standard_concept	concept_code	valid_start_date	valid_end_date	invalid_reaso
#219661	40232756	oxycodone hydrochloride 5 MG Oral Tablet	Drug	RxNorm	        Clinical Drug	            S	        1049621	        20110102	        20991231	        NaN

OHSDI:
The concept code represents the identifier of the Concept in the source vocabulary, such as SNOMED-CT concept IDs, RxNorm RXCUIs etc. Note that concept codes are not unique across vocabularies.


con[con.concept_id == 2000010348] # ?None
con[con.concept_id == 723020] 
#316375	723020	diazepam 10 MG Oral Tablet	Drug	RxNorm	Clinical Drug	S	197589	19700101	20991231	NaN


con.concept_id.value_counts().describe() # ok, all 1, this is a primary key.
'''


con.head()


### Patient 1: 3912882389848878631
On se concentre sur un premier cas avec 24 lignes pour 1 personne: 
Attention, c'est un integer, pas un texte. Ne pas mettre de ''


In [ ]:
patient1 = drug_exposure.loc[drug_exposure['person_id'] == 3912882389848878631, liste_non_null]
'''
patient1.columns
Index(['drug_exposure_id', 'person_id', 'drug_concept_id',
       'drug_exposure_start_date', 'drug_exposure_start_datetime',
       'drug_exposure_end_date', 'drug_exposure_end_datetime',
       'drug_type_concept_id', 'quantity', 'route_concept_id',
       'visit_occurrence_id', 'drug_source_value', 'drug_source_concept_id',
       'route_source_value', 'dose_unit_source_value'],
      dtype='object')
'''
patient1

In [117]:
patient = pd.merge(patient1, con, how='left', left_on = 'drug_concept_id', right_on = 'concept_id')

#patient.shape, patient1.shape, con.shape #((24, 25), (24, 15), (516708, 10))
#patient

In [126]:
patient = patient.sort_values(by='drug_exposure_start_datetime')
number_types = patient.dose_unit_source_value.describe()['unique']

fig3 = make_subplots(rows = 1, cols = number_types, subplot_titles=(patient.dose_unit_source_value.value_counts().index))
colonne = 1

for drug_type in patient.dose_unit_source_value.value_counts().index:
    #print(drug_type)

    for d in patient[patient.dose_unit_source_value == drug_type].concept_name.unique():
        
        temp = patient[patient.concept_name == d]
        nom_drug = str(d)[0:50] # I hav to limit because some are too long... ut How to be sue we have everything needed ?

        #print(temp.dose_unit_source_value.unique())
        fig3.add_trace(go.Scatter(x = temp.drug_exposure_start_datetime, y = temp.quantity, name=f'{nom_drug}, {temp.dose_unit_source_value.unique()}'), row=1, col=colonne)

    colonne += 1

fig3.update_layout(title=dict(text=f'Person_id: {patient.person_id[0]}', font=dict(size=30)))



fig3.show()

In [132]:
# TOGGLE GRAPH - not useful

patient = patient.sort_values(by='drug_exposure_start_datetime')
number_types = patient.dose_unit_source_value.describe()['unique']

#fig3 = make_subplots(rows = 1, cols = number_types, subplot_titles=(patient.dose_unit_source_value.value_counts().index))
colonne = 1

for drug_type in patient.dose_unit_source_value.value_counts().index:
    #print(drug_type)
    traces = []
    buttons = []

    # for each drug type, create a liste of traces and buttons
    for d in patient[patient.dose_unit_source_value == drug_type].concept_id.unique():
        temp = patient[patient.concept_id == d]
        nom_drug = str(d)[0:50]

        traces.append(go.Scatter(x = temp.drug_exposure_start_datetime, y = temp.quantity, name=f'{d}, {temp.dose_unit_source_value.unique()}', visible = True))
        
        buttons.append(dict(method='update',
                        label=f'{d}',
                        visible=True,
                        args=[{'visible':[x == d for x in patient[patient.dose_unit_source_value == drug_type].concept_id.unique()]}],
                        args2=[{'visible':[x != d for x in patient[patient.dose_unit_source_value == drug_type].concept_id.unique()]}]
                        )
                )

    # Make the graph with toggle
        
    # add a button 'All' to toggle them all together    
    layout = go.Layout(
        updatemenus=[
            dict(
                type='buttons',
                direction='right',
                x=0.7,
                y=1.3,
                showactive=True,
                buttons=buttons
            )
        ],
        showlegend=True

    )

    fig3 = go.Figure(data=traces,layout=layout)
    fig3.update_layout(xaxis=dict(rangeslider=dict(autorange=True, range=[patient.drug_exposure_start_datetime.min(), patient.drug_exposure_start_datetime.max()])))
   
    fig3.show()




# https://plotly.com/python/range-slider/ axes avec plusieurs type d'échelle

In [116]:
patient1[patient1.dose_unit_source_value == 'TAB']
# 9 lignes, des exposure_id différentes, avec des dates différents (2129 ????)
# drug_concept_id being the name of the drug I guess, Quantity in Tablettes.


drug_exposure_id            person_id  drug_concept_id  \
5622  6996991851384754967  3912882389848878631         40232756   
5623 -2528282390097198867  3912882389848878631         40232756   
5624  8598302712376844332  3912882389848878631         40232756   
5625  4557254971515693852  3912882389848878631         40232756   
5635 -6600566077470540957  3912882389848878631         21125875   
5636  8139119374042756604  3912882389848878631         19137328   
5639 -1076779580276771548  3912882389848878631          1127433   
5640 -4804305780328125843  3912882389848878631           723020   
5641 -2002420539696298274  3912882389848878631         19077513   

     drug_exposure_start_date drug_exposure_start_datetime  \
5622               2129-01-05          2129-01-05 00:00:00   
5623               2129-01-04          2129-01-04 18:00:00   
5624               2129-01-05          2129-01-05 02:00:00   
5625               2129-01-04          2129-01-04 21:00:00   
5635               2129-01-05          2129-01-05 08:00:00   
5636               2129-01-05          2129-01-05 08:00:00   
5639               2129-01-03          2129-01-03 20:00:00   
5640               2129-01-03          2129-01-03 19:00:00   
5641               2129-01-05          2129-01-05 08:00:00   

     drug_exposure_end_date drug_exposure_end_datetime  drug_type_concept_id  \
5622             2129-01-05        2129-01-05 01:00:00                 32838   
5623             2129-01-04        2129-01-04 23:00:00                 32838   
5624             2129-01-05        2129-01-05 02:00:00                 32838   
5625             2129-01-04        2129-01-04 23:00:00                 32838   
5635             2129-01-05        2129-01-05 18:00:00                 32838   
5636             2129-01-05        2129-01-05 18:00:00                 32838   
5639             2129-01-05        2129-01-05 18:00:00                 32838   
5640             2129-01-05        2129-01-05 02:00:00                 32838   
5641             2129-01-05        2129-01-05 18:00:00                 32838   

      quantity  route_concept_id  visit_occurrence_id drug_source_value  \
5622       1.5           4167540  3761046692024108622       00406055262   
5623       0.5           4167540  3761046692024108622       00406055262   
5624       1.0           4167540  3761046692024108622       00406055262   
5625       1.0           4167540  3761046692024108622       00406055262   
5635       1.0           4167540  3761046692024108622       00904053061   
5636       1.0           4167540  3761046692024108622       00904054460   
5639       2.0           4167540  3761046692024108622       51079000220   
5640       1.0           4167540  3761046692024108622       51079028620   
5641       1.0           4167540  3761046692024108622       62584089701   

      drug_source_concept_id route_source_value dose_unit_source_value  
5622                45266765              PO/NG                    TAB  
5623                45266765              PO/NG                    TAB  
5624                45266765              PO/NG                    TAB  
5625                45266765              PO/NG                    TAB  
5635                45165542              PO/NG                    TAB  
5636                45114473              PO/NG                    TAB  
5639                45325177              PO/NG                    TAB  
5640                45000972              PO/NG                    TAB  
5641                44954855              PO/NG                    TAB

In [5]:
patient1_drug1 = patient1.loc[patient1.drug_concept_id==40232756,['drug_exposure_start_datetime', 'drug_exposure_end_datetime', 'quantity']].sort_values(by='drug_exposure_start_datetime')



patient1_drug1['cumulative'] = [0.5, 1.5, 3.0, 4.0]



# ne fonctionne pas comme ça, car les date ne sont pas correct.... il faut prendre datetime déjà, puis trier en fonction du temps.
# il faut faire avec plotly un graph pour choisir le medoc.
# il faut calculer le cumulatif ou non ?
patient1_drug1


drug_exposure_start_datetime drug_exposure_end_datetime  quantity  \
5623          2129-01-04 18:00:00        2129-01-04 23:00:00       0.5   
5625          2129-01-04 21:00:00        2129-01-04 23:00:00       1.0   
5622          2129-01-05 00:00:00        2129-01-05 01:00:00       1.5   
5624          2129-01-05 02:00:00        2129-01-05 02:00:00       1.0   

      cumulative  
5623         0.5  
5625         1.5  
5622         3.0  
5624         4.0

In [63]:
patient1_drug1.drug_exposure_start_datetime.min()

'2129-01-04 18:00:00'

In [ ]:
result = pd.merge(left, right, on="key")

In [25]:

fig = make_subplots(rows = 1, cols = 2, subplot_titles=('not cumulative','cumulative'))
fig.add_trace(go.Scatter(x = patient1_drug1.drug_exposure_start_datetime, y = patient1_drug1.quantity), row=1, col=1)
fig.update_layout(xaxis=dict(rangeslider=dict(autorange=True, range=[patient1_drug1.drug_exposure_start_datetime.min(), patient1_drug1.drug_exposure_start_datetime.max()])))
   
fig.add_trace(go.Scatter(x = patient1_drug1.drug_exposure_start_datetime, y = patient1_drug1.cumulative), row=1, col=2)
  
fig.show()

In [12]:
# Graph with concept_id:

patient1 = patient1.sort_values(by='drug_exposure_start_datetime')
number_types = patient1.dose_unit_source_value.describe()['unique']

fig3 = make_subplots(rows = 1, cols = number_types, subplot_titles=(patient1.dose_unit_source_value.value_counts().index))
colonne = 1

for drug_type in patient1.dose_unit_source_value.value_counts().index:
    #print(drug_type)

    for d in patient1[patient1.dose_unit_source_value == drug_type].drug_concept_id.unique():
        temp = patient1[patient1.drug_concept_id == d]
        #print(temp.dose_unit_source_value.unique())
        fig3.add_trace(go.Scatter(x = temp.drug_exposure_start_datetime, y = temp.quantity, name=f'{d}, {temp.dose_unit_source_value.unique()}'), row=1, col=colonne)

    colonne += 1

fig3.show()

# Add curseur de temps, add legendgroup, add selection de chaque drug -> drug selection is already doable as well as time

In [113]:
patient1_drugName.loc[patient1_drugName.concept_name == d,'drug_concept_id']

18    723020
Name: drug_concept_id, dtype: int64

In [13]:
# TOGGLE GRAPH - not useful

patient1 = patient1.sort_values(by='drug_exposure_start_datetime')
number_types = patient1.dose_unit_source_value.describe()['unique']

#fig3 = make_subplots(rows = 1, cols = number_types, subplot_titles=(patient1.dose_unit_source_value.value_counts().index))
colonne = 1

for drug_type in patient1.dose_unit_source_value.value_counts().index:
    #print(drug_type)
    traces = []
    buttons = []

    # for each drug type, create a liste of traces and buttons
    for d in patient1[patient1.dose_unit_source_value == drug_type].drug_concept_id.unique():
        temp = patient1[patient1.drug_concept_id == d]

        traces.append(go.Scatter(x = temp.drug_exposure_start_datetime, y = temp.quantity, name=f'{d}, {temp.dose_unit_source_value.unique()}', visible = True))
        
        buttons.append(dict(method='update',
                        label=f'{d}',
                        visible=True,
                        args=[{'visible':[x == d for x in patient1[patient1.dose_unit_source_value == drug_type].drug_concept_id.unique()]}],
                        args2=[{'visible':[x != d for x in patient1[patient1.dose_unit_source_value == drug_type].drug_concept_id.unique()]}]
                        )
                )

    # Make the graph with toggle
        
    # add a button 'All' to toggle them all together    
    layout = go.Layout(
        updatemenus=[
            dict(
                type='buttons',
                direction='right',
                x=0.7,
                y=1.3,
                showactive=True,
                buttons=buttons
            )
        ],
        showlegend=True

    )

    fig3 = go.Figure(data=traces,layout=layout)
    fig3.update_layout(xaxis=dict(rangeslider=dict(autorange=True, range=[patient1_drug1.drug_exposure_start_datetime.min(), patient1_drug1.drug_exposure_start_datetime.max()])))
   
    fig3.show()




# https://plotly.com/python/range-slider/ axes avec plusieurs type d'échelle

### Drug_exposure start and end can be different

In [ ]:
# drug_exposure.drug_exposure_start_datetime.value_counts()
#drug_exposure.drug_exposure_end_datetime.value_counts()
print(drug_exposure.drug_exposure_start_datetime[0], drug_exposure.drug_exposure_end_datetime[0])
# Can be different....

# Dose Era

A Dose Era is defined as a span of time when the Person is assumed to be exposed to a constant dose of a specific active ingredient.

Dose Form information is not taken into account. So, if the patient changes between different formulations, or different manufacturers with the same formulation, the Dose Era is still spanning the entire time of exposure to the Ingredient.

In [15]:
dose_era = pd.read_csv('Data/dose_era.csv')
dose_era.head()

#dose_era.shape #(117, 7)

dose_era_id            person_id  drug_concept_id  unit_concept_id  \
0 -1101936257632052351  5548892236933978704           902427             8576   
1  9002582707624403415  4985579811051920670           714785             8576   
2  5495900697391314629 -4873075614181207858          1328165             8576   
3  6335084700122752427 -8205283012979532608          1328165             8576   
4  2848105832570024513 -4873075614181207858          1328165             8576   

   dose_value dose_era_start_date dose_era_end_date  
0        10.0          2142-05-14        2142-05-16  
1         2.5          2110-04-13        2110-04-14  
2       120.0          2148-09-15        2148-09-15  
3       120.0          2189-06-11        2189-06-13  
4       120.0          2147-12-20        2147-12-21

In [13]:
dose_era.unit_concept_id.value_counts()
# 8576 are all in mg.... This is good :) I can go back to drug_concept id, dose_value....

unit_concept_id
8576    117
Name: count, dtype: int64

In [16]:
dose_era.drug_concept_id.value_counts()

drug_concept_id
1307046    38
1328165    12
718583     12
1383815    11
745466     10
965748      8
750982      7
1398937     4
1503297     2
1560171     2
1163944     2
1318853     2
1154029     2
714785      1
1353776     1
719311      1
1353766     1
902427      1
Name: count, dtype: int64

# Drug era

**active ingredient**, successive period of drug_exposure to the same ingredent gives drug era

A Drug Era is defined as a span of time when the Person is assumed to be exposed to a particular active ingredient. A Drug Era is not the same as a Drug Exposure:
Exposures are individual records corresponding to the source when Drug was delivered to the Person, while successive periods of Drug Exposures are combined under certain rules to produce continuous Drug Eras.

-> not the same concept_id because here we talk about an ingredient, not a particular drug... diazepam vs diazepam tablet 10mg.  

Example:  
drug_era: person 3912882389848878631 got 2 exposure to diazepam 723013 between 2129-01-03	2129-01-05  
[6598	-7540355434127132424	3912882389848878631	723013	2129-01-03	2129-01-05	2	0]   

drug_exposure: person 3912882389848878631 got 1 drug_exposure to 723020 within 2129-01-03 19:00:00	and 2129-01-05 02:00:00	  
[5640	-4804305780328125843	3912882389848878631	723020	2129-01-03	2129-01-03 19:00:00	2129-01-05	2129-01-05 02:00:00	NaN	32838	NaN	...	NaN	4167540	NaN	NaN	3761046692024108622	NaN	51079028620	45000972	PO/NG	TAB]

In [14]:
dose_era[dose_era.person_id == 3912882389848878631] # my patient is not recognized, neither my drug_cncept_id...why ????
#patient1[patient1.dose_unit_source_value == drug_type]
dose_era[dose_era.drug_concept_id == 19127213]
# ne pas mettre de guillements, ça ne fonctionne pas...

Empty DataFrame
Columns: [dose_era_id, person_id, drug_concept_id, unit_concept_id, dose_value, dose_era_start_date, dose_era_end_date]
Index: []

In [4]:
drug_era = pd.read_csv('Data/drug_era.csv')

drug_era.shape #(7931, 7)
drug_era.head()

drug_era_id            person_id  drug_concept_id  \
0 -4731852071843461961 -2067961723109232727          1124957   
1  1725818565432188772  -626229666378242477          1368671   
2  3222075720408667327  -626229666378242477           967823   
3  7411613425123823172   579254014084392336          1310149   
4  4265287408545791371 -2312013739856114142         19095164   

  drug_era_start_date drug_era_end_date  drug_exposure_count  gap_days  
0          2169-01-21        2169-01-24                    1         0  
1          2171-11-11        2171-11-16                    2         1  
2          2171-11-11        2171-11-22                   12         0  
3          2115-10-16        2115-10-18                    2         0  
4          2116-06-28        2116-07-05                    1         0

In [ ]:
drug_era.drug_concept_id.value_counts()

In [5]:
drug_era[drug_era.drug_concept_id == 19127213] # cannot be found...

drug_era[drug_era.person_id == 3912882389848878631] # -> ok...

drug_era_id            person_id  drug_concept_id  \
6596   500595522397761829  3912882389848878631         43531944   
6597  6330107922049835930  3912882389848878631          1125315   
6598 -7540355434127132424  3912882389848878631           723013   
6599  7165454045076608944  3912882389848878631         40166605   
6600 -8180642557726588733  3912882389848878631         40164828   
6601  3025930947898329035  3912882389848878631         42873956   
6602  8800459384878548729  3912882389848878631           967823   
6603  -338966274758502286  3912882389848878631          1367571   
6683  1590324940605174808  3912882389848878631           923645   
6684  8866077140124362516  3912882389848878631          1124957   
6690 -1112804010670705686  3912882389848878631         36878782   
6691 -2016335706505619184  3912882389848878631         19111620   
6692  2426943909930076013  3912882389848878631          1560524   
6693  8939824964893766208  3912882389848878631         19137312   
6694  4416439696488798469  3912882389848878631          1177480   
6695  -945752478111843518  3912882389848878631           734275   

     drug_era_start_date drug_era_end_date  drug_exposure_count  gap_days  
6596          2129-01-03        2129-01-05                    1         0  
6597          2129-01-03        2129-01-05                    1         0  
6598          2129-01-03        2129-01-05                    2         0  
6599          2129-01-03        2129-01-05                    1         0  
6600          2129-01-03        2129-01-05                    1         0  
6601          2129-01-03        2129-01-05                    1         0  
6602          2129-01-03        2129-01-05                    2         0  
6603          2129-01-03        2129-01-05                    1         0  
6683          2129-01-04        2129-01-05                    1         0  
6684          2129-01-04        2129-01-05                    4         1  
6690          2129-01-05        2129-01-05                    1         0  
6691          2129-01-05        2129-01-05                    1         0  
6692          2129-01-05        2129-01-05                    1         0  
6693          2129-01-05        2129-01-05                    1         0  
6694          2129-01-05        2129-01-05                    1         0  
6695          2129-01-05        2129-01-11                    7         6

In [ ]:
temp_1 = drug_era.loc[drug_era.person_id == 3912882389848878631, 'drug_concept_id'].unique() #16
temp_1.sort()
'''
drug_era
(array([  723013 - 	diazepam as an ingredient,   734275,   923645,   967823,  1124957,  1125315,
         1177480,  1367571,  1560524, 19111620, 19137312, 36878782,
        40164828, 40166605, 42873956, 43531944]),

drug_exposure
 array([  723020 - diazepam 10 MG Oral Tablet RxNorm,   734281,  1127433, 19019050, 19019418, 19077513,
        19079250, 19126251, 19127213, 19137328, 21125875, 36249736,
        40221329, 40232756, 43011850, 45775137]))
'''
temp_2 = drug_exposure.loc[drug_exposure.person_id == 3912882389848878631, 'drug_concept_id'].unique() #16
temp_2.sort()

# all the same

temp_1, temp_2


In [10]:
temp_3 = dose_era.loc[dose_era.person_id == 3912882389848878631]
temp_3 # none
#temp_3.loc[temp_3.drug_concept_id == 723013]

Empty DataFrame
Columns: [dose_era_id, person_id, drug_concept_id, unit_concept_id, dose_value, dose_era_start_date, dose_era_end_date]
Index: []

In [ ]:
temp_exp = drug_exposure.person_id.unique()
len(temp_exp) #100
temp_drug = drug_era.person_id.unique()
len(temp_drug) #100
temp_dose = dose_era.person_id.unique()
len(temp_dose) #44

# drug_strength

In [17]:
drug_strength = pd.read_csv('Data/DRUG_STRENGTH.csv', sep='\s+')
drug_strength.shape #(200983, 12)

'''
Columns (5) have mixed types. Specify dtype option on import or set low_memory=False.


drug_concept_id                  int64
ingredient_concept_id            int64
amount_value                   float64
amount_unit_concept_id           int64
numerator_value                float64
numerator_unit_concept_id       object
denominator_value              float64
denominator_unit_concept_id    float64
box_size                       float64
valid_start_date               float64
valid_end_date                 float64
invalid_reason                 float64
'''

drug_strength.head()

/tmp/ipykernel_5721/1970751477.py:1: DtypeWarning:

Columns (5) have mixed types. Specify dtype option on import or set low_memory=False.



drug_concept_id  ingredient_concept_id  amount_value  \
0          1154535                1154343         0.417   
1          1154536                1154343         0.210   
2          1731598                1731597         2.000   
3          1396092                1396012       225.000   
4          1517764                1517740         0.002   

   amount_unit_concept_id  numerator_value numerator_unit_concept_id  \
0                    8576           8587.0                  19700101   
1                    8576           8587.0                  19700101   
2                    8576           8587.0                  19700101   
3                    8576       19700101.0                  20991231   
4                    8576           8587.0                  19700101   

   denominator_value  denominator_unit_concept_id  box_size  valid_start_date  \
0         20991231.0                          NaN       NaN               NaN   
1         20991231.0                          NaN       NaN               NaN   
2         20991231.0                          NaN       NaN               NaN   
3                NaN                          NaN       NaN               NaN   
4         20991231.0                          NaN       NaN               NaN   

   valid_end_date  invalid_reason  
0             NaN             NaN  
1             NaN             NaN  
2             NaN             NaN  
3             NaN             NaN  
4             NaN             NaN

In [27]:
drug_strength.info()
'''
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 200983 entries, 0 to 200982
Data columns (total 12 columns):
 #   Column                       Non-Null Count   Dtype  
---  ------                       --------------   -----  
 0   drug_concept_id              200983 non-null  int64  
 1   ingredient_concept_id        200983 non-null  int64  
 2   amount_value                 200983 non-null  float64
 3   amount_unit_concept_id       200983 non-null  int64  
 4   numerator_value              200983 non-null  float64
 5   numerator_unit_concept_id    142454 non-null  object 
 6   denominator_value            79593 non-null   float64
 7   denominator_unit_concept_id  7763 non-null    float64
 8   box_size                     0 non-null       float64
 9   valid_start_date             0 non-null       float64
 10  valid_end_date               0 non-null       float64
 11  invalid_reason               0 non-null       float64
dtypes: float64(8), int64(3), object(1)
memory usage: 18.4+ MB
'''


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 200983 entries, 0 to 200982
Data columns (total 12 columns):
 #   Column                       Non-Null Count   Dtype  
---  ------                       --------------   -----  
 0   drug_concept_id              200983 non-null  int64  
 1   ingredient_concept_id        200983 non-null  int64  
 2   amount_value                 200983 non-null  float64
 3   amount_unit_concept_id       200983 non-null  int64  
 4   numerator_value              200983 non-null  float64
 5   numerator_unit_concept_id    142454 non-null  object 
 6   denominator_value            79593 non-null   float64
 7   denominator_unit_concept_id  7763 non-null    float64
 8   box_size                     0 non-null       float64
 9   valid_start_date             0 non-null       float64
 10  valid_end_date               0 non-null       float64
 11  invalid_reason               0 non-null       float64
dtypes: float64(8), int64(3), object(1)
memory usage: 18.4+ MB


In [29]:
drug_strength[drug_strength['drug_concept_id']==734281] # it works
'''	
phenobarbital 65 MG/ML

drug_concept_id	ingredient_concept_id	amount_value	amount_unit_concept_id	numerator_value	numerator_unit_concept_id	denominator_value	denominator_unit_concept_id	box_size	valid_start_date	valid_end_date	invalid_reason
734281	        734275	                65.0	            8576	                8587.0	    19700101	                20991231.0	            NaN	NaN	NaN	NaN	NaN
8576 = mg... should be mg/ml
8587 = mL...numeratot ! That's why. check with a TAB and others, check the numerator_unit_conceptid
'''

drug_concept_id  ingredient_concept_id  amount_value  \
38328           734281                 734275          65.0   

       amount_unit_concept_id  numerator_value numerator_unit_concept_id  \
38328                    8576           8587.0                  19700101   

       denominator_value  denominator_unit_concept_id  box_size  \
38328         20991231.0                          NaN       NaN   

       valid_start_date  valid_end_date  invalid_reason  
38328               NaN             NaN             NaN

In [32]:
drug_strength.numerator_value.unique() 
drug_strength.amount_unit_concept_id.unique()

array([    8576,     8510,     9551, 45744811,     8587,     9573,
       45744812, 45744810,     9693, 44819154, 45744813,    32018,
          32407, 45744814,     9278,     9439,     9367, 45744816,
           8718,     8554, 19700101, 20050911, 20080302, 20080330,
       20080629, 20080831, 20080928, 20090201, 20220103, 20070128,
       20080427, 20070429, 20060514, 20060226, 20071230, 20050724,
       20210405, 20230103, 20220801, 20060709, 20051204, 20070729,
       20060402, 20190603, 20180205, 20060115, 20070225, 20051113,
       20080601, 20080127, 20180402, 20070527, 20070701, 20200803,
       20060312, 20070902, 20070401, 20060730, 20090104, 20190204,
       20060827, 20070114, 20190903, 20190401, 20190701, 20180702,
       20190506, 20221003, 20210802, 20180806, 20061119, 20061217,
       20170807, 20170103, 20220307, 20070930, 20090503, 20081130,
       20090531, 20080727, 20081109, 20060604, 20090301, 20061008,
       20181203, 20160502, 20210503, 20160801, 20210104, 20200

# Other RxNorm files

In [3]:
#temp = pd.read_csv('../Downloads/vocabulary_download_v5_{e387ec98-436f-4ec5-9c97-5da7c14b445d}_1710335286650/CONCEPT_ANCESTOR.csv', sep='\s+')
'''
# CONCEPT_ANCESTOR.csv _ don't care, this is:
ancestor_concept_id	descendant_concept_id	min_levels_of_separation	max_levels_of_separation
0	742267	40172924	2	3
1	701449	1525417	1	1

  CONCEPT_CLASS.csv  _
  concept_class_id	concept_class_name	concept_class_concept_id
0	Qualifier Value	Qualifier Value	44819021
1	Quality Metric	Quality Metric	44819092
  
  CONCEPT.csv  
  concept_id	concept_name	domain_id	vocabulary_id	concept_class_id	standard_concept	concept_code	valid_start_date	valid_end_date	invalid_reason
0	1146945	concept.concept_id	Metadata	CDM	Field	S	CDM1	20141111	20991231	NaN
1	1146954	concept.invalid_reason	Metadata	CDM	Field	S	CDM10	20141111	20991231	
shape (516708, 10)

concept[concept.concept_name == 'numerator_value'] # cannot find it

  CONCEPT_RELATIONSHIP.csv  
  concept_id_1	concept_id_2	relationship_id	valid_start_date	valid_end_date	invalid_reason
0	42628856	42628856	Mapped from	20160606	20991231	NaN
1	42628856	42628856	Maps to	20160606	20991231	NaN

  CONCEPT_SYNONYM.csv  
  concept_id	concept_synonym_name	language_concept_id
0	19000382	Lactobacillus bulgaricus	4180186
1	19000521	Lariam 250 MG Oral Tablet	4180186
(113673, 3)

  DOMAIN.csv sep='\t'
domain_id	domain_name	domain_concept_id
0	Cost	Cost	581456
1	Condition/Device	Condition/Device	235
2	Gender	Gender	2

 DRUG_STRENGTH.csv  \s+
 
 RELATIONSHIP.csv  
 relationship_id	relationship_name	is_hierarchical	defines_ancestry	reverse_relationship_id	relationship_concept_id
0	Has specimen proc	Has specimen procedure (SNOMED)	0	0	Specimen proc of	44818775
1	Has specimen source	Has specimen source identity (SNOMED)	0	0	Specimen identity of	44818776
2	Has specimen morph	Has specimen source morphology (SNOMED)	0	0	Specimen morph of	44818777
 
 VOCABULARY.csv
vocabulary_id	vocabulary_name	vocabulary_reference	vocabulary_version	vocabulary_concept_id
0	SOPT	Source of Payment Typology (PHDSC)	https://www.nahdo.org/sopt	SOPT Version 9.2	32473
1	Episode	OMOP Episode	OMOP generated	Episode 20201014	32523
2	Observation Type	OMOP Observation Type	OMOP generated	NaN	44819129
shape (38, 5)


'''


liste = ['CONCEPT_CLASS.csv', 'CONCEPT.csv','CONCEPT_RELATIONSHIP.csv','CONCEPT_SYNONYM.csv','DOMAIN.csv','RELATIONSHIP.csv', 'VOCABULARY.csv']

for fich in liste:
    print(fich)
    temp = pd.read_csv(f'../Downloads/vocabulary_RxNorm/{fich}', sep='\t')
    display(temp.head())

CONCEPT_CLASS.csv


concept_class_id concept_class_name  concept_class_concept_id
0  Qualifier Value    Qualifier Value                  44819021
1   Quality Metric     Quality Metric                  44819092
2             Race               Race                  44819030
3             Read               Read                  44819273
4  Record Artifact    Record Artifact                  44818999

CONCEPT.csv


/tmp/ipykernel_20524/246075802.py:34: DtypeWarning: Columns (6,9) have mixed types. Specify dtype option on import or set low_memory=False.
  temp = pd.read_csv(f'../Downloads/vocabulary_RxNorm/{fich}', sep='\t')


concept_id                              concept_name domain_id  \
0     1146945                        concept.concept_id  Metadata   
1     1146954                    concept.invalid_reason  Metadata   
2     1147044  observation_period.observation_period_id  Metadata   
3      756315         metadata.metadata_type_concept_id  Metadata   
4      756316                             metadata.name  Metadata   

  vocabulary_id concept_class_id standard_concept concept_code  \
0           CDM            Field                S         CDM1   
1           CDM            Field                S        CDM10   
2           CDM            Field                S       CDM100   
3           CDM            Field                S      CDM1000   
4           CDM            Field                S      CDM1001   

   valid_start_date  valid_end_date invalid_reason  
0          20141111        20991231            NaN  
1          20141111        20991231            NaN  
2          20141111        20991231            NaN  
3          20210925        20991231            NaN  
4          20210925        20991231            NaN

CONCEPT_RELATIONSHIP.csv


concept_id_1  concept_id_2 relationship_id  valid_start_date  \
0      42628856      42628856     Mapped from          20160606   
1      42628856      42628856         Maps to          20160606   
2      42628856      42629622    Tradename of          20160606   
3      42628856      45776706     Constitutes          20160606   
4      42628857       1318137  RxNorm has ing          20160606   

   valid_end_date  invalid_reason  
0        20991231             NaN  
1        20991231             NaN  
2        20991231             NaN  
3        20991231             NaN  
4        20991231             NaN

CONCEPT_SYNONYM.csv


concept_id                        concept_synonym_name  language_concept_id
0    19000382                    Lactobacillus bulgaricus              4180186
1    19000521                   Lariam 250 MG Oral Tablet              4180186
2    19000811               Lactobacillus casei rhamnosus              4180186
3    19000821                            sulfaisodimidine              4180186
4    19000824  magaldrate 800 MG per 5 ML Oral Suspension              4180186

DOMAIN.csv


domain_id       domain_name  domain_concept_id
0              Cost              Cost             581456
1  Condition/Device  Condition/Device                235
2            Gender            Gender                  2
3              Race              Race                  3
4         Ethnicity         Ethnicity                  4

RELATIONSHIP.csv


relationship_id                        relationship_name  \
0    Has specimen proc          Has specimen procedure (SNOMED)   
1  Has specimen source    Has specimen source identity (SNOMED)   
2   Has specimen morph  Has specimen source morphology (SNOMED)   
3    Has specimen topo  Has specimen source topography (SNOMED)   
4   Has specimen subst          Has specimen substance (SNOMED)   

   is_hierarchical  defines_ancestry reverse_relationship_id  \
0                0                 0        Specimen proc of   
1                0                 0    Specimen identity of   
2                0                 0       Specimen morph of   
3                0                 0        Specimen topo of   
4                0                 0       Specimen subst of   

   relationship_concept_id  
0                 44818775  
1                 44818776  
2                 44818777  
3                 44818778  
4                 44818779

VOCABULARY.csv


vocabulary_id                     vocabulary_name  \
0              SOPT  Source of Payment Typology (PHDSC)   
1           Episode                        OMOP Episode   
2  Observation Type               OMOP Observation Type   
3    Procedure Type      OMOP Procedure Occurrence Type   
4         Meas Type               OMOP Measurement Type   

         vocabulary_reference vocabulary_version  vocabulary_concept_id  
0  https://www.nahdo.org/sopt   SOPT Version 9.2                  32473  
1              OMOP generated   Episode 20201014                  32523  
2              OMOP generated                NaN               44819129  
3              OMOP generated                NaN               44819128  
4              OMOP generated                NaN               44819152

In [15]:
#fich = ['CONCEPT.csv']
concept = pd.read_csv('../Downloads/vocabulary_RxNorm/CONCEPT.csv', sep='\t')
concept.shape

/tmp/ipykernel_20524/2293929783.py:2: DtypeWarning: Columns (6,9) have mixed types. Specify dtype option on import or set low_memory=False.
  concept = pd.read_csv('../Downloads/vocabulary_RxNorm/CONCEPT.csv', sep='\t')


(516708, 10)

In [17]:
concept[concept.concept_name == 'amount_unit_concept_id'] # cannot find it

Empty DataFrame
Columns: [concept_id, concept_name, domain_id, vocabulary_id, concept_class_id, standard_concept, concept_code, valid_start_date, valid_end_date, invalid_reason]
Index: []

In [4]:
syn = pd.read_csv(f'../Downloads/vocabulary_RxNorm/CONCEPT_SYNONYM.csv', sep='\t')
syn.head()

concept_id                        concept_synonym_name  language_concept_id
0    19000382                    Lactobacillus bulgaricus              4180186
1    19000521                   Lariam 250 MG Oral Tablet              4180186
2    19000811               Lactobacillus casei rhamnosus              4180186
3    19000821                            sulfaisodimidine              4180186
4    19000824  magaldrate 800 MG per 5 ML Oral Suspension              4180186

In [10]:
syn.loc[syn.concept_id == 723020] # None
syn.loc[syn.language_concept_id == 44819021] # None
syn.loc[syn.concept_id == 8576] # None
syn.loc[syn.language_concept_id == 8576] # None
syn.loc[syn.concept_id == 8576] # None
syn.loc[syn.concept_id == 40232756] # None
'''
8576	mg	milligram	Unit	Standard	Valid	Unit	UCUM
'''

Empty DataFrame
Columns: [concept_id, concept_synonym_name, language_concept_id]
Index: []

In [65]:
syn.loc[syn.concept_id == 40232756] # None


Empty DataFrame
Columns: [concept_id, concept_synonym_name, language_concept_id]
Index: []

## 2b_concept, 2b_concept_relationship
5.2.2 Concept Names

Each concept has one name. Names are always in English. They are imported from the source of the vocabulary. If the source vocabulary has more than one name, the most expressive is selected and the remaining ones are stored in the CONCEPT_SYNONYM table under the same CONCEPT_ID key. Non-English names are recorded in CONCEPT_SYNONYM as well, with the appropriate language concept ID in the LANGUAGE_CONCEPT_ID field. The name is 255 characters long, which means that very long names get truncated and the full-length version recorded as another synonym, which can hold up to 1000 characters.  


measurement = 1 measrment/1 person at a certain time.... Not OK !

START_DATE, END_DATE, INVALID_REASON are refering to the drug and the name of the drug. Not its expiration date.

**Each concept is assigned a domain in the DOMAIN_ID field**, which in contrast to the numerical CONCEPT_ID is a short case-sensitive unique alphanumeric ID for the domain.


In [12]:
liste = ['2b_concept.csv','cohort_attribute.csv','death.csv','location.csv','observation_period.csv','visit_detail.csv',
         '2b_concept_relationship.csv','cohort.csv','device_exposure.csv','measurement.csv','payer_plan_period.csv','visit_occurrence.csv',
         '2b_vocabulary.csv','cohort_definition.csv','dose_era.csv','metadata.csv','person.csv',
         'attribute_definition.csv','condition_era.csv','drug_era.csv','note.csv','procedure_occurrence.csv',
         'care_site.csv','condition_occurrence.csv','drug_exposure.csv','note_nlp.csv','provider.csv',
         'cdm_source.csv','cost.csv','fact_relationship.csv','observation.csv','specimen.csv']

for fich in liste:
    print(fich)
    temp = pd.read_csv(f'../Downloads/Data/{fich}')#, sep='\t')
    display(temp.head())

# measurement -> units ?
# 2b_concept -W Drug name ?

2b_concept.csv


concept_id                     concept_name domain_id     vocabulary_id  \
0  2000011360       Caffeine Citrate 1 Syringe      Drug  mimiciv_drug_ndc   
1  2000010536                    Acetaminophen      Drug  mimiciv_drug_ndc   
2  2000011213                   Melatonin 1 MG      Drug  mimiciv_drug_ndc   
3  2000011355  0.9% Sodium Chloride 100 mL Bag      Drug  mimiciv_drug_ndc   
4  2000010313                       Vancomycin      Drug  mimiciv_drug_ndc   

    concept_class_id standard_concept                          concept_code  \
0  Prescription Drug              NaN            Caffeine Citrate 1 Syringe   
1  Prescription Drug              NaN               ACETAMINOPHEN (RECTAL)    
2  Prescription Drug              NaN                           melatonin 1   
3  Prescription Drug              NaN       0.9% Sodium Chloride 100 mL Bag   
4  Prescription Drug              NaN  Vancomycin ophthalmic 50mg/ml bottle   

  valid_start_DATE valid_end_DATE  invalid_reason  
0       1970-01-01     2099-12-31             NaN  
1       1970-01-01     2099-12-31             NaN  
2       1970-01-01     2099-12-31             NaN  
3       1970-01-01     2099-12-31             NaN  
4       1970-01-01     2099-12-31             NaN

cohort_attribute.csv


Empty DataFrame
Columns: [cohort_definition_id, subject_id, cohort_start_date, cohort_end_date, attribute_definition_id, value_as_number, value_as_concept_id]
Index: []

death.csv


person_id  death_date       death_datetime  \
0 -2312013739856114142  2116-07-05  2116-07-05 08:05:00   
1 -7671795861352464589  2115-10-12  2115-10-12 00:00:00   
2  1194579079287927665  2177-03-29  2177-03-29 14:15:00   
3   579254014084392336  2117-03-24  2117-03-24 00:01:00   
4 -4353160957725823366  2146-07-12  2146-07-12 00:00:00   

   death_type_concept_id  cause_concept_id  cause_source_value  \
0                  32817                 0                 NaN   
1                  32817                 0                 NaN   
2                  32817                 0                 NaN   
3                  32817                 0                 NaN   
4                  32817                 0                 NaN   

   cause_source_concept_id  
0                        0  
1                        0  
2                        0  
3                        0  
4                        0

location.csv


location_id  address_1  address_2  city state  zip  county  \
0            1        NaN        NaN   NaN    MA  NaN     NaN   

  location_source_value  
0  Beth Israel Hospital

observation_period.csv


observation_period_id            person_id observation_period_start_date  \
0    -422211212329812262 -7391666713304457659                    2110-11-30   
1   -4759444374091930779 -8254164865273971123                    2150-03-05   
2    2583710258594541919 -8205283012979532608                    2189-06-09   
3   -8621531794134526280  8090044958540695372                    2143-03-10   
4   -6927702067122852728  3589912774911670296                    2153-09-15   

  observation_period_end_date  period_type_concept_id  
0                  2110-12-10                   32828  
1                  2151-01-25                   32828  
2                  2189-06-13                   32828  
3                  2148-02-05                   32828  
4                  2153-09-25                   32828

visit_detail.csv


visit_detail_id            person_id  visit_detail_concept_id  \
0 -1757828362327778468  3129727379702505063                     8870   
1 -4357165027259445573  3129727379702505063                     8870   
2  8055254568125344353  3129727379702505063                   581385   
3 -9202240471022785074  1741351032930224901                    32037   
4 -3849978153133518996  1741351032930224901                     8870   

  visit_detail_start_date visit_detail_start_datetime visit_detail_end_date  \
0              2197-04-17         2197-04-17 09:48:00            2197-04-17   
1              2197-04-16         2197-04-16 22:57:00            2197-04-17   
2              2197-04-16         2197-04-16 22:57:00            2197-04-16   
3              2177-07-14         2177-07-14 20:38:00            2177-07-15   
4              2177-07-14         2177-07-14 14:52:00            2177-07-14   

  visit_detail_end_datetime  visit_detail_type_concept_id  provider_id  \
0       2197-04-17 11:44:19                         32817          NaN   
1       2197-04-17 09:48:00                         32817          NaN   
2       2197-04-16 22:57:00                         32817          NaN   
3       2177-07-15 16:08:36                         32817          NaN   
4       2177-07-14 20:38:00                         32817          NaN   

   care_site_id  admitting_source_concept_id  discharge_to_concept_id  \
0 -3.633440e+18                       8870.0                      NaN   
1  6.888076e+18                       8870.0                      NaN   
2           NaN                       8870.0                      NaN   
3  7.883370e+18                       8870.0                 581476.0   
4  6.888076e+18                       8870.0                  32037.0   

   preceding_visit_detail_id   visit_detail_source_value  \
0                        NaN  10002930|25282382|38481760   
1                        NaN  10002930|25282382|35169671   
2                        NaN           10002930|25282382   
3              -3.849978e+18  10037928|22490490|30229866   
4                        NaN  10037928|22490490|35750269   

   visit_detail_source_concept_id admitting_source_value  \
0                      2000001903         EMERGENCY ROOM   
1                      2000001901         EMERGENCY ROOM   
2                      2000001805         EMERGENCY ROOM   
3                      2000001622   Emergency Department   
4                      2000001901         EMERGENCY ROOM   

                          discharge_to_source_value  visit_detail_parent_id  \
0                                               NaN                     NaN   
1                                               NaN                     NaN   
2                                               NaN                     NaN   
3                                  HOME HEALTH CARE                     NaN   
4  Medical/Surgical Intensive Care Unit (MICU/SICU)                     NaN   

   visit_occurrence_id  
0 -9127810274408915712  
1 -9127810274408915712  
2 -9127810274408915712  
3  3736965967695233281  
4  3736965967695233281

2b_concept_relationship.csv


concept_id_1  concept_id_2 relationship_id valid_start_DATE valid_end_DATE  \
0    2000003069       4022792         Maps to       1970-01-01     2099-12-31   
1    2000010663      40164921         Maps to       1970-01-01     2099-12-31   
2    2000020189      38001595         Maps to       1970-01-01     2099-12-31   
3    2000020866      38001445         Maps to       1970-01-01     2099-12-31   
4    2000020336      38001223         Maps to       1970-01-01     2099-12-31   

   invalid_reason  
0             NaN  
1             NaN  
2             NaN  
3             NaN  
4             NaN

cohort.csv


Empty DataFrame
Columns: [cohort_definition_id, subject_id, cohort_start_date, cohort_end_date]
Index: []

device_exposure.csv


device_exposure_id           person_id  device_concept_id  \
0 -6080797302205697410 -626229666378242477           45768171   
1 -5973025877998015677 -626229666378242477           45768171   
2 -8106791820473225988 -626229666378242477           45768171   
3  4432515289879643590 -626229666378242477           45768171   
4  1718665515813261089 -626229666378242477           45768171   

  device_exposure_start_date device_exposure_start_datetime  \
0                 2171-11-14            2171-11-14 13:00:00   
1                 2171-11-14            2171-11-14 20:00:00   
2                 2171-11-14            2171-11-14 11:28:00   
3                 2171-11-14            2171-11-14 17:00:00   
4                 2171-11-14            2171-11-14 15:00:00   

  device_exposure_end_date device_exposure_end_datetime  \
0               2171-11-14          2171-11-14 13:00:00   
1               2171-11-14          2171-11-14 20:00:00   
2               2171-11-14          2171-11-14 11:28:00   
3               2171-11-14          2171-11-14 17:00:00   
4               2171-11-14          2171-11-14 15:00:00   

   device_type_concept_id  unique_device_id  quantity  provider_id  \
0                   32817               NaN       NaN          NaN   
1                   32817               NaN       NaN          NaN   
2                   32817               NaN       NaN          NaN   
3                   32817               NaN       NaN          NaN   
4                   32817               NaN       NaN          NaN   

   visit_occurrence_id  visit_detail_id  device_source_value  \
0 -1486256937339039377              NaN               224087   
1 -1486256937339039377              NaN               224087   
2 -1486256937339039377              NaN               224087   
3 -1486256937339039377              NaN               224087   
4 -1486256937339039377              NaN               224087   

   device_source_concept_id  
0                2000030021  
1                2000030021  
2                2000030021  
3                2000030021  
4                2000030021

measurement.csv


/tmp/ipykernel_11458/1342704.py:10: DtypeWarning: Columns (16) have mixed types. Specify dtype option on import or set low_memory=False.
  temp = pd.read_csv(f'../Downloads/Data/{fich}')#, sep='\t')


measurement_id            person_id  measurement_concept_id  \
0  7620661609057829801 -7437341330444582833                 3007913   
1 -6166868866082303206 -2312013739856114142                 3012501   
2 -5240588523649662838 -4234372750442829205                 3012501   
3  6455538495061268502  8805478484003283429                 3012501   
4  8972684215776493449  6339505631013617478                 3012501   

  measurement_date measurement_datetime  measurement_time  \
0       2113-09-14  2113-09-14 10:41:00               NaN   
1       2116-07-05  2116-07-05 05:51:00               NaN   
2       2154-01-02  2154-01-02 20:18:00               NaN   
3       2114-06-20  2114-06-20 09:59:00               NaN   
4       2111-11-15  2111-11-15 03:09:00               NaN   

   measurement_type_concept_id  operator_concept_id  value_as_number  \
0                        32856                  NaN            586.0   
1                        32856                  NaN             -4.0   
2                        32856                  NaN             -2.0   
3                        32856                  NaN              3.0   
4                        32856                  NaN             -2.0   

   value_as_concept_id  unit_concept_id  range_low  range_high  provider_id  \
0                  NaN           8876.0        NaN         NaN          NaN   
1                  NaN           9557.0        NaN         NaN          NaN   
2                  NaN           9557.0        NaN         NaN          NaN   
3                  NaN           9557.0        NaN         NaN          NaN   
4                  NaN           9557.0        NaN         NaN          NaN   

   visit_occurrence_id  visit_detail_id measurement_source_value  \
0  3697313480337443666              NaN                    50801   
1 -5005846256467230136              NaN                    50802   
2 -5317261811030552609              NaN                    50802   
3  4821424463988433938              NaN                    50802   
4 -6354047184485090226              NaN                    50802   

   measurement_source_concept_id unit_source_value value_source_value  
0                   2.000001e+09             mm Hg                586  
1                   2.000001e+09             mEq/L                 -4  
2                   2.000001e+09             mEq/L                 -2  
3                   2.000001e+09             mEq/L                  3  
4                   2.000001e+09             mEq/L                 -2

payer_plan_period.csv


Empty DataFrame
Columns: [payer_plan_period_id, person_id, payer_plan_period_start_date, payer_plan_period_end_date, payer_concept_id, payer_source_value, payer_source_concept_id, plan_concept_id, plan_source_value, plan_source_concept_id, sponsor_concept_id, sponsor_source_value, sponsor_source_concept_id, family_source_value, stop_reason_concept_id, stop_reason_source_value, stop_reason_source_concept_id]
Index: []

visit_occurrence.csv


visit_occurrence_id            person_id  visit_concept_id  \
0 -4406053801395356975  4783904755296699562          38004207   
1  2636026522589494723 -6225647829918357531          38004207   
2 -8900473297538187901  7918537411740862407          38004207   
3 -8091470952174089624 -6225647829918357531          38004207   
4  8144493519470334167  7155255168997124770          38004207   

  visit_start_date visit_start_datetime visit_end_date   visit_end_datetime  \
0       2112-11-06  2112-11-06 11:05:00     2112-11-06  2112-11-06 11:05:00   
1       2153-10-17  2153-10-17 14:23:00     2153-10-17  2153-10-17 14:23:00   
2       2130-10-10  2130-10-10 16:20:00     2130-10-10  2130-10-10 16:20:00   
3       2155-04-20  2155-04-20 09:00:00     2155-04-20  2155-04-20 09:00:00   
4       2147-07-08  2147-07-08 08:20:00     2147-07-08  2147-07-08 08:20:00   

   visit_type_concept_id  provider_id  care_site_id   visit_source_value  \
0                  32817          NaN           NaN  10035631|2112-11-06   
1                  32817          NaN           NaN  10019003|2153-10-17   
2                  32817          NaN           NaN  10020306|2130-10-10   
3                  32817          NaN           NaN  10019003|2155-04-20   
4                  32817          NaN           NaN  10014354|2147-07-08   

   visit_source_concept_id  admitting_source_concept_id  \
0               2000001801                          NaN   
1               2000001801                          NaN   
2               2000001801                          NaN   
3               2000001801                          NaN   
4               2000001801                          NaN   

  admitting_source_value  discharge_to_concept_id discharge_to_source_value  \
0                    NaN                      NaN                       NaN   
1                    NaN                      NaN                       NaN   
2                    NaN                      NaN                       NaN   
3                    NaN                      NaN                       NaN   
4                    NaN                      NaN                       NaN   

   preceding_visit_occurrence_id  
0                  -3.100296e+18  
1                  -2.238366e+18  
2                  -8.600570e+18  
3                   4.782862e+18  
4                  -3.275690e+18

2b_vocabulary.csv


vocabulary_id                 vocabulary_name  \
0          mimiciv_micro_organism          mimiciv_micro_organism   
1              mimiciv_drug_route              mimiciv_drug_route   
2           mimiciv_obs_insurance           mimiciv_obs_insurance   
3  mimiciv_cond_chartevents_value  mimiciv_cond_chartevents_value   
4  mimiciv_vis_admission_location  mimiciv_vis_admission_location   

  vocabulary_reference  vocabulary_version  vocabulary_concept_id  
0   Odysseus generated                 NaN             2110000014  
1   Odysseus generated                 NaN             2110000005  
2   Odysseus generated                 NaN             2110000019  
3   Odysseus generated                 NaN             2110000002  
4   Odysseus generated                 NaN             2110000025

cohort_definition.csv


Empty DataFrame
Columns: [cohort_definition_id, cohort_definition_name, cohort_definition_description, definition_type_concept_id, cohort_definition_syntax, subject_concept_id, cohort_initiation_date]
Index: []

dose_era.csv


dose_era_id            person_id  drug_concept_id  unit_concept_id  \
0 -1101936257632052351  5548892236933978704           902427             8576   
1  9002582707624403415  4985579811051920670           714785             8576   
2  5495900697391314629 -4873075614181207858          1328165             8576   
3  6335084700122752427 -8205283012979532608          1328165             8576   
4  2848105832570024513 -4873075614181207858          1328165             8576   

   dose_value dose_era_start_date dose_era_end_date  
0        10.0          2142-05-14        2142-05-16  
1         2.5          2110-04-13        2110-04-14  
2       120.0          2148-09-15        2148-09-15  
3       120.0          2189-06-11        2189-06-13  
4       120.0          2147-12-20        2147-12-21

metadata.csv


Empty DataFrame
Columns: [metadata_concept_id, metadata_type_concept_id, name, value_as_string, value_as_concept_id, metadata_date, metadata_datetime]
Index: []

person.csv


person_id  gender_concept_id  year_of_birth  month_of_birth  \
0  3589912774911670296               8507           2095             NaN   
1 -3210373572193940939               8507           2079             NaN   
2  -775517641933593374               8507           2149             NaN   
3 -2575767131279873665               8507           2050             NaN   
4 -8970844422700220177               8507           2114             NaN   

   day_of_birth  birth_datetime  race_concept_id  ethnicity_concept_id  \
0           NaN             NaN                0              38003563   
1           NaN             NaN                0              38003563   
2           NaN             NaN             8516                     0   
3           NaN             NaN             8516                     0   
4           NaN             NaN             8527                     0   

   location_id  provider_id  care_site_id  person_source_value  \
0          NaN          NaN           NaN             10009628   
1          NaN          NaN           NaN             10011398   
2          NaN          NaN           NaN             10004235   
3          NaN          NaN           NaN             10024043   
4          NaN          NaN           NaN             10038933   

  gender_source_value  gender_source_concept_id       race_source_value  \
0                   M                         0                     NaN   
1                   M                         0                     NaN   
2                   M                         0  BLACK/AFRICAN AMERICAN   
3                   M                         0  BLACK/AFRICAN AMERICAN   
4                   M                         0                   WHITE   

   race_source_concept_id ethnicity_source_value  ethnicity_source_concept_id  
0                       0        HISPANIC/LATINO                   2000001408  
1                       0        HISPANIC/LATINO                   2000001408  
2              2000001406                    NaN                            0  
3              2000001406                    NaN                            0  
4              2000001404                    NaN                            0

attribute_definition.csv


Empty DataFrame
Columns: [attribute_definition_id, attribute_name, attribute_description, attribute_type_concept_id, attribute_syntax]
Index: []

condition_era.csv


condition_era_id            person_id  condition_concept_id  \
0 -4017843510320986981 -8254164865273971123               4140598   
1 -7667301639797895423  7131048714591189903                434610   
2 -7455077195684470206  7131048714591189903                201826   
3 -4223886229831787734  7131048714591189903                437827   
4 -8340460082370905381  7131048714591189903                 27674   

  condition_era_start_date condition_era_end_date  condition_occurrence_count  
0               2150-03-11             2150-04-04                         256  
1               2189-05-22             2189-05-25                           1  
2               2189-05-22             2189-05-25                           1  
3               2189-05-22             2189-05-25                           1  
4               2189-05-22             2189-05-25                           1

drug_era.csv


drug_era_id            person_id  drug_concept_id  \
0 -4731852071843461961 -2067961723109232727          1124957   
1  1725818565432188772  -626229666378242477          1368671   
2  3222075720408667327  -626229666378242477           967823   
3  7411613425123823172   579254014084392336          1310149   
4  4265287408545791371 -2312013739856114142         19095164   

  drug_era_start_date drug_era_end_date  drug_exposure_count  gap_days  
0          2169-01-21        2169-01-24                    1         0  
1          2171-11-11        2171-11-16                    2         1  
2          2171-11-11        2171-11-22                   12         0  
3          2115-10-16        2115-10-18                    2         0  
4          2116-06-28        2116-07-05                    1         0

note.csv


Empty DataFrame
Columns: [note_id, person_id, note_date, note_datetime, note_type_concept_id, note_class_concept_id, note_title, note_text, encoding_concept_id, language_concept_id, provider_id, visit_occurrence_id, visit_detail_id, note_source_value]
Index: []

procedure_occurrence.csv


procedure_occurrence_id            person_id  procedure_concept_id  \
0     -6348795981381799385  4783904755296699562               2102720   
1      7881544392229438243  7918537411740862407               2102732   
2     -5408721103708766509  4498126063475867818               2104913   
3      5368606506786851886  8090044958540695372              42742532   
4     -8177080120515460365  5863607150722936210               2108296   

  procedure_date   procedure_datetime  procedure_type_concept_id  \
0     2113-07-18  2113-07-18 14:55:00                      32821   
1     2129-10-30  2129-10-30 13:20:00                      32821   
2     2157-07-18  2157-07-18 16:49:00                      32821   
3     2143-03-19  2143-03-19 12:00:00                      32821   
4     2142-07-07  2142-07-07 16:40:00                      32821   

   modifier_concept_id  quantity  provider_id  visit_occurrence_id  \
0                    0       1.0          NaN  -433474223361412760   
1                    0       1.0          NaN  7730200099818586525   
2                    0       1.0          NaN  2514722389640694708   
3                    0       1.0          NaN  6734319289435533605   
4                    0       1.0          NaN -5935156580867756808   

   visit_detail_id procedure_source_value  procedure_source_concept_id  \
0              NaN                  19301                      2102720   
1              NaN                  19303                      2102732   
2              NaN                  27235                      2104913   
3              NaN                  36252                     42742532   
4              NaN                  36830                      2108296   

   modifier_source_value  
0                    NaN  
1                    NaN  
2                    NaN  
3                    NaN  
4                    NaN

care_site.csv


care_site_id       care_site_name  place_of_service_concept_id  \
0  7482024270134467590         Med/Surg/GYN                          NaN   
1  6872771148744744676  Trauma SICU (TSICU)                          NaN   
2  4662258935512879919              Unknown                          NaN   
3 -2501038380264497908             Med/Surg                          NaN   
4  8603684584038321926          Observation                          NaN   

   location_id care_site_source_value place_of_service_source_value  
0            1           Med/Surg/GYN                  Med/Surg/GYN  
1            1    Trauma SICU (TSICU)           Trauma SICU (TSICU)  
2            1                Unknown                       Unknown  
3            1               Med/Surg                      Med/Surg  
4            1            Observation                   Observation

condition_occurrence.csv


condition_occurrence_id            person_id  condition_concept_id  \
0      7000818053728441484  1741351032930224901                196523   
1     -3514320024333679102  1741351032930224901                436659   
2     -6297447354934110463  1741351032930224901                435515   
3      3131132784874295309  1741351032930224901                436096   
4      2246709778332812674  1741351032930224901                440383   

  condition_start_date condition_start_datetime condition_end_date  \
0           2179-07-24      2179-07-24 18:21:00         2179-07-28   
1           2179-07-24      2179-07-24 18:21:00         2179-07-28   
2           2179-07-24      2179-07-24 18:21:00         2179-07-28   
3           2179-07-24      2179-07-24 18:21:00         2179-07-28   
4           2179-07-24      2179-07-24 18:21:00         2179-07-28   

  condition_end_datetime  condition_type_concept_id  stop_reason  provider_id  \
0    2179-07-28 15:54:00                      32821          NaN          NaN   
1    2179-07-28 15:54:00                      32821          NaN          NaN   
2    2179-07-28 15:54:00                      32821          NaN          NaN   
3    2179-07-28 15:54:00                      32821          NaN          NaN   
4    2179-07-28 15:54:00                      32821          NaN          NaN   

   visit_occurrence_id  visit_detail_id condition_source_value  \
0 -5779522865065417426              NaN                78791     
1 -5779522865065417426              NaN                2809      
2 -5779522865065417426              NaN                2761      
3 -5779522865065417426              NaN                33829     
4 -5779522865065417426              NaN                311       

   condition_source_concept_id  condition_status_source_value  \
0                     44824628                            NaN   
1                     44828816                            NaN   
2                     44822952                            NaN   
3                     44834629                            NaN   
4                     44833421                            NaN   

   condition_status_concept_id  
0                          NaN  
1                          NaN  
2                          NaN  
3                          NaN  
4                          NaN

drug_exposure.csv


drug_exposure_id            person_id  drug_concept_id  \
0   294884377115777655  1741351032930224901         40166274   
1 -3609243742606366340  1741351032930224901         40166274   
2 -6865345241721388581  1741351032930224901         40166274   
3  -826223020394544622  1741351032930224901         40166274   
4  2417954811860157314  1741351032930224901         40166274   

  drug_exposure_start_date drug_exposure_start_datetime  \
0               2177-07-16          2177-07-16 22:00:00   
1               2177-07-17          2177-07-17 19:00:00   
2               2177-07-15          2177-07-15 19:00:00   
3               2177-07-21          2177-07-21 22:00:00   
4               2177-07-18          2177-07-18 23:00:00   

  drug_exposure_end_date drug_exposure_end_datetime  verbatim_end_date  \
0             2177-07-17        2177-07-17 21:00:00                NaN   
1             2177-07-18        2177-07-18 18:00:00                NaN   
2             2177-07-16        2177-07-16 18:00:00                NaN   
3             2177-07-22        2177-07-22 21:00:00                NaN   
4             2177-07-19        2177-07-19 22:00:00                NaN   

   drug_type_concept_id  stop_reason  ...  sig  route_concept_id  lot_number  \
0                 32838          NaN  ...  NaN           4142048         NaN   
1                 32838          NaN  ...  NaN           4142048         NaN   
2                 32838          NaN  ...  NaN           4142048         NaN   
3                 32838          NaN  ...  NaN           4142048         NaN   
4                 32838          NaN  ...  NaN           4142048         NaN   

   provider_id  visit_occurrence_id  visit_detail_id  drug_source_value  \
0          NaN  3736965967695233281              NaN        00002751001   
1          NaN  3736965967695233281              NaN        00002751001   
2          NaN  3736965967695233281              NaN        00002751001   
3          NaN  3736965967695233281              NaN        00002751001   
4          NaN  3736965967695233281              NaN        00002751001   

   drug_source_concept_id  route_source_value dose_unit_source_value  
0                45144375                  SC                   VIAL  
1                45144375                  SC                   VIAL  
2                45144375                  SC                   VIAL  
3                45144375                  SC                   VIAL  
4                45144375                  SC                   VIAL  

[5 rows x 23 columns]

note_nlp.csv


Empty DataFrame
Columns: [note_nlp_id, note_id, section_concept_id, snippet, offset, lexical_variant, note_nlp_concept_id, note_nlp_source_concept_id, nlp_system, nlp_date, nlp_datetime, term_exists, term_temporal, term_modifiers]
Index: []

provider.csv


Empty DataFrame
Columns: [provider_id, provider_name, npi, dea, specialty_concept_id, care_site_id, year_of_birth, gender_concept_id, provider_source_value, specialty_source_value, specialty_source_concept_id, gender_source_value, gender_source_concept_id]
Index: []

cdm_source.csv


cdm_source_name cdm_source_abbreviation cdm_holder  \
0        MIMIC IV                 mimiciv  PhysioNet   

                                  source_description  \
0  MIMIC-IV is a publicly available database of p...   

   source_documentation_reference                cdm_etl_reference  \
0  https://mimic-iv.mit.edu/docs/  https://github.com/OHDSI/MIMIC/   

  source_release_date cdm_release_date cdm_version vocabulary_version  
0          2020-09-01       2021-05-03       5.3.1     v5.0 10-SEP-20

cost.csv


Empty DataFrame
Columns: [cost_id, cost_event_id, cost_domain_id, cost_type_concept_id, currency_concept_id, total_charge, total_cost, total_paid, paid_by_payer, paid_by_patient, paid_patient_copay, paid_patient_coinsurance, paid_patient_deductible, paid_by_primary, paid_ingredient_cost, paid_dispensing_fee, payer_plan_period_id, amount_allowed, revenue_code_concept_id, revenue_code_source_value, drg_concept_id, drg_source_value]
Index: []

[0 rows x 22 columns]

fact_relationship.csv


domain_concept_id_1            fact_id_1  domain_concept_id_2  \
0                   21 -1528338048118240227                   21   
1                   21 -4413670399601597669                   21   
2                   21    62647581672888119                   21   
3                   21  8612006927123919607                   21   
4                   21  5642509002092905641                   21   

             fact_id_2  relationship_concept_id  
0 -2626995764797704161                   581436  
1 -5035119046096715237                   581436  
2  7870255279990651748                   581436  
3 -2294747223347562162                   581436  
4    59077947031565658                   581436

observation.csv


observation_id            person_id  observation_concept_id  \
0 -7221215846302395378  7131048714591189903                 4215685   
1 -9113461341871004757  7131048714591189903                  440922   
2 -6378179537712623650  7131048714591189903                44784283   
3 -3000731179192887483  7131048714591189903                 4296248   
4 -8633553265894813739  7131048714591189903                 4296248   

  observation_date observation_datetime  observation_type_concept_id  \
0       2189-05-22  2189-05-22 23:18:00                        32821   
1       2189-05-22  2189-05-22 23:18:00                        32821   
2       2189-05-22  2189-05-22 23:18:00                        32821   
3       2189-05-22  2189-05-22 23:18:00                        32817   
4       2189-05-22  2189-05-22 23:18:00                        32817   

   value_as_number                       value_as_string  value_as_concept_id  \
0              NaN                                   NaN                  NaN   
1              NaN                                   NaN                  NaN   
2              NaN                                   NaN                  NaN   
3              NaN  Septicemia & Disseminated Infections                  0.0   
4              NaN  Septicemia & Disseminated Infections                  0.0   

   qualifier_concept_id  unit_concept_id  provider_id  visit_occurrence_id  \
0                   NaN              NaN          NaN  1832418602869683763   
1                   NaN              NaN          NaN  1832418602869683763   
2                   NaN              NaN          NaN  1832418602869683763   
3                   NaN              NaN          NaN  1832418602869683763   
4                   NaN              NaN          NaN  1832418602869683763   

   visit_detail_id observation_source_value  observation_source_concept_id  \
0              NaN                  V4972                         44827368   
1              NaN                  V5867                         44820462   
2              NaN                  V0481                         44837741   
3              NaN                     7203                              0   
4              NaN                     7203                              0   

   unit_source_value  qualifier_source_value  
0                NaN                     NaN  
1                NaN                     NaN  
2                NaN                     NaN  
3                NaN                     NaN  
4                NaN                     NaN

specimen.csv


specimen_id            person_id  specimen_concept_id  \
0 -5102033398575528989  4668337230155062633              4001183   
1  5035924384215166531  2288881942133868955              4001183   
2  4680974717267536012  3192038106523208432              4001183   
3  1036293910445291253  3192038106523208432              4001183   
4  4665676571104609028  3192038106523208432              4001183   

   specimen_type_concept_id specimen_date    specimen_datetime  quantity  \
0                     32856    2117-07-16  2117-07-16 10:00:00       NaN   
1                     32856    2157-11-20  2157-11-20 12:20:00       NaN   
2                     32856    2136-04-19  2136-04-19 14:33:00       NaN   
3                     32856    2136-05-05  2136-05-05 11:00:00       NaN   
4                     32856    2136-04-27  2136-04-27 00:00:00       NaN   

   unit_concept_id  anatomic_site_concept_id  disease_status_concept_id  \
0              NaN                         0                          0   
1              NaN                         0                          0   
2              NaN                         0                          0   
3              NaN                         0                          0   
4              NaN                         0                          0   

                                  specimen_source_id  specimen_source_value  \
0  {"subject_id":10021487,"hadm_id":20429160,"mic...                  70003   
1  {"subject_id":10001217,"hadm_id":24597018,"mic...                  70003   
2  {"subject_id":10007795,"hadm_id":28477357,"mic...                  70003   
3  {"subject_id":10007795,"hadm_id":25135483,"mic...                  70003   
4  {"subject_id":10007795,"hadm_id":null,"microev...                  70003   

   unit_source_value  anatomic_site_source_value  disease_status_source_value  
0                NaN                         NaN                          NaN  
1                NaN                         NaN                          NaN  
2                NaN                         NaN                          NaN  
3                NaN                         NaN                          NaN  
4                NaN                         NaN                          NaN

In [14]:
fich = 'measurement.csv'
mea = pd.read_csv(f'../Downloads/Data/{fich}')#, sep='\t')
print(mea.shape) #(338550, 20)

fich = '2b_concept.csv'
twob_con = pd.read_csv(f'../Downloads/Data/{fich}')#, sep='\t')
print(twob_con.shape) #(3885, 10)

# measurement -> units ?
# 2b_concept -W Drug name ?

/tmp/ipykernel_11458/2246964919.py:2: DtypeWarning: Columns (16) have mixed types. Specify dtype option on import or set low_memory=False.
  mea = pd.read_csv(f'../Downloads/Data/{fich}')#, sep='\t')


In [25]:
mea.head() # measurment is 1 measurment.... so depends on people and so.... I can use the unit_source_value alright, but nothing to o with what I want !!!!
mea.loc[mea.unit_concept_id == 8576] # Nope

mea.unit_source_value.value_counts() # ok, but not the dictionnary as I want

unit_source_value
mmHg        58321
%           33430
mEq/L       23145
mg/dL       22902
insp/min    21782
            ...  
hrs             1
GPL             1
MPL             1
mL/min          1
U/g/Hb          1
Name: count, Length: 71, dtype: int64

In [17]:
twob_con.head()


(3885, 10)


concept_id                     concept_name domain_id     vocabulary_id  \
0  2000011360       Caffeine Citrate 1 Syringe      Drug  mimiciv_drug_ndc   
1  2000010536                    Acetaminophen      Drug  mimiciv_drug_ndc   
2  2000011213                   Melatonin 1 MG      Drug  mimiciv_drug_ndc   
3  2000011355  0.9% Sodium Chloride 100 mL Bag      Drug  mimiciv_drug_ndc   
4  2000010313                       Vancomycin      Drug  mimiciv_drug_ndc   

    concept_class_id standard_concept                          concept_code  \
0  Prescription Drug              NaN            Caffeine Citrate 1 Syringe   
1  Prescription Drug              NaN               ACETAMINOPHEN (RECTAL)    
2  Prescription Drug              NaN                           melatonin 1   
3  Prescription Drug              NaN       0.9% Sodium Chloride 100 mL Bag   
4  Prescription Drug              NaN  Vancomycin ophthalmic 50mg/ml bottle   

  valid_start_DATE valid_end_DATE  invalid_reason  
0       1970-01-01     2099-12-31             NaN  
1       1970-01-01     2099-12-31             NaN  
2       1970-01-01     2099-12-31             NaN  
3       1970-01-01     2099-12-31             NaN  
4       1970-01-01     2099-12-31             NaN

# Dico / Nom des médoc:
2b_concept.
Mais je n'ai pas leur concept_id ok.... why ????

In [34]:
#twob_con.loc[twob_con.concept_id = 723020] # None

twob_con.concept_id.value_counts().describe() # 1 everywhere. concept_id being the primary key.... But doesn't fit with my values earlier, like 723020.*
twob_con.concept_name.value_counts().describe() # not 1 everywhere -> several medoc. medicine .
twob_con[twob_con.concept_name == 'Vancomycin'] # concept_id being the primary key.... But doesn't fit with my values earlier, like 723020.

'''
SOMETIMES, SOME ARE THE SAME? BUT WRITTEN DIFFERENTLY
	concept_id	concept_name	domain_id	vocabulary_id	concept_class_id	standard_concept	concept_code	valid_start_DATE	valid_end_DATE	invalid_reason
4	2000010313	Vancomycin	Drug	mimiciv_drug_ndc	Prescription Drug	NaN	Vancomycin ophthalmic 50mg/ml bottle	1970-01-01	2099-12-31	NaN
134	2000011142	Vancomycin	Drug	mimiciv_drug_ndc	Prescription Drug	NaN	Vancomycin fortified opthalmic 50mg/ml	1970-01-01	2099-12-31	NaN
211	2000011334	Vancomycin	Drug	mimiciv_drug_ndc	Prescription Drug	NaN	Vancomycin ophthalmic solution 50mg/ml	1970-01-01	2099-12-31	NaN
411	2000010294	Vancomycin	Drug	mimiciv_drug_ndc	Prescription Drug	NaN	vancoMYCIN for nasal inhalation 200 mg / 2 mL	1970-01-01	2099-12-31	NaN
1020	2000010348	Vancomycin	Drug	mimiciv_drug_ndc	Prescription Drug	NaN	vancoMYCIN for inhalation 200 mg / 2 mL	1970-01-01	2099-12-31	NaN
1086	2000010020	Vancomycin	Drug	mimiciv_drug_ndc	Prescription Drug	NaN	Vancomycin HCl 1GM FROZ. BAG	1970-01-01	2099-12-31	NaN
1371	2000010621	Vancomycin	Drug	mimiciv_drug_ndc	Prescription Drug	NaN	Vancomycin Enema 200mg/100ml ns	1970-01-01	2099-12-31	NaN
1384	2000011343	Vancomycin	Drug	mimiciv_drug_ndc	Prescription Drug	NaN	Vancomycin Enema Vancomycin 1mg/ml	1970-01-01	2099-12-31	NaN

concept name - medoc de départ
concept code - nom donné dans les BDD


'''

#Ok. Cool le nom des médoc', mais pourquoi, je n'ai pas leur concept_id...

count    3885.0
mean        1.0
std         0.0
min         1.0
25%         1.0
50%         1.0
75%         1.0
max         1.0
Name: count, dtype: float64

In [38]:
fich = '2b_concept_relationship.csv'
twob_con_rel = pd.read_csv(f'../Downloads/Data/{fich}')#, sep='\t')
print(twob_con_rel.shape) #(7716, 6)
twob_con_rel

twob_con_rel.info() #concept_id        3885 non-null   int64

(7716, 6)


concept_id_1  concept_id_2 relationship_id valid_start_DATE  \
0       2000003069       4022792         Maps to       1970-01-01   
1       2000010663      40164921         Maps to       1970-01-01   
2       2000020189      38001595         Maps to       1970-01-01   
3       2000020866      38001445         Maps to       1970-01-01   
4       2000020336      38001223         Maps to       1970-01-01   
...            ...           ...             ...              ...   
7711       4078442    2000004053     Mapped from       1970-01-01   
7712       4078442    2000004125     Mapped from       1970-01-01   
7713       4078442    2000004083     Mapped from       1970-01-01   
7714       4078442    2000004015     Mapped from       1970-01-01   
7715       4078442    2000004096     Mapped from       1970-01-01   

     valid_end_DATE  invalid_reason  
0        2099-12-31             NaN  
1        2099-12-31             NaN  
2        2099-12-31             NaN  
3        2099-12-31             NaN  
4        2099-12-31             NaN  
...             ...             ...  
7711     2099-12-31             NaN  
7712     2099-12-31             NaN  
7713     2099-12-31             NaN  
7714     2099-12-31             NaN  
7715     2099-12-31             NaN  

[7716 rows x 6 columns]

In [67]:
twob_con_rel.loc[twob_con_rel.concept_id_1 == 2000010348] 
''' 
    concept_id_1	concept_id_2	relationship_id	valid_start_DATE	valid_end_DATE	invalid_reason
872	2000010348	    1707687	        Maps to	        1970-01-01	        2099-12-31      	NaN
'''
twob_con_rel.loc[twob_con_rel.concept_id_2 == 40232756] # 1707687 being the ingredient


Empty DataFrame
Columns: [concept_id_1, concept_id_2, relationship_id, valid_start_DATE, valid_end_DATE, invalid_reason]
Index: []

In [71]:
twob_con_rel.info() #concept_id        3885 non-null   int64

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7716 entries, 0 to 7715
Data columns (total 6 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   concept_id_1      7716 non-null   int64  
 1   concept_id_2      7716 non-null   int64  
 2   relationship_id   7716 non-null   object 
 3   valid_start_DATE  7716 non-null   object 
 4   valid_end_DATE    7716 non-null   object 
 5   invalid_reason    0 non-null      float64
dtypes: float64(1), int64(2), object(3)
memory usage: 361.8+ KB


In [66]:
twob_con_rel.loc[twob_con_rel.concept_id_1 == 40232756]  # None

Empty DataFrame
Columns: [concept_id_1, concept_id_2, relationship_id, valid_start_DATE, valid_end_DATE, invalid_reason]
Index: []

In [49]:

twob_con_rel.concept_id_1.value_counts() # several/number

concept_id_1
4078442       73
4001225       16
4171047       15
968426        13
4141149       12
              ..
2000010662     1
2000026270     1
2000011212     1
2000026279     1
2000020750     1
Name: count, Length: 6593, dtype: int64

In [50]:
twob_con_rel.concept_id_2.value_counts() # several/number

concept_id_2
4078442       73
4001225       16
4171047       15
968426        13
4141149       12
              ..
38001575       1
1748959        1
38001591       1
705944         1
2000004096     1
Name: count, Length: 6593, dtype: int64